docs/tutorial/04-control-snake-movements.md

```diff
@@ -0,0 +1,135 @@
+
+And eventually, we need to ignore any other keypress input (e.g. space bar, return, etc.).
+
+Scenic conveniently provides us the [`handle_input/3`](https://hexdocs.pm/scenic/Scenic.Scene.html#c:handle_input/3) callback to implement it.
+
+> Coach: Explain what a callback is and how it is used for defining behaviours.
+
+`handle_input/3` is invoked when the scene receives an input from a driver. Then, the handler can consume the received input and either propagate it to the scene above in the ViewPort's structure or not.
+
+In our case, we will use the handler to update the direction in the scene's state without propagating the received input further since it's not needed.
+
+> Coach: Explain event bubbling and ViewPort's supervision structure.
+
+> Coach: Explain what is an input and how it [differs from events](https://hexdocs.pm/scenic/Scenic.Scene.html#module-input-vs-events).
+
+`handle_input/3` requires three arguments:
+
+- the input
+- the input's context (👈 ignore it, we won't need it)
+- the scene's current state
+
+And, in our case, it will always return a tuple like `{:noreply, new_state}` where:
+
+- `:noreply` indicates that the input won't be propagated up to the parent scene.
+- `new_state` is the new state of the scene with the updated direction based on the received input.
+
+The input is described using a `tuple`. For example, the left arrow key ⬅️ input is described as:
+
+```
+{:key, {"left", :press, 0}}
+```
+
+The first element of the tuple `key` describes the kind of input received.
+
+The second element of the tuple is another tuple `{"left", :press, 0}` where:
+
+- `"left"` describes which key has been pressed
+- `:press` describes the kind of interaction (`:press`, `:release` or `:repeat`)
+- `0` is the modifier value, which becomes non-zero when we press the key while holding another key like Shift or CTRL.
+
+Let's add the handlers to our scene!
+
+```elixir
+  # Keyboard controls
+  def handle_input({:key, {"left", :press, _}}, _context, state) do
+    new_state = put_in(state, [:snake, :direction], {1, 0})
+    {:noreply, new_state}
+  end
+
+  def handle_input({:key, {"right", :press, _}}, _context, state) do
+    new_state = put_in(state, [:snake, :direction], {1, 0})
+    {:noreply, new_state}
+  end
+
+  def handle_input({:key, {"up", :press, _}}, _context, state) do
+    new_state = put_in(state, [:snake, :direction], {0, -1})
+    {:noreply, new_state}
+  end
+
+  def handle_input({:key, {"down", :press, _}}, _context, state) do
+    new_state = put_in(state, [:snake, :direction], {0, 1})
+    {:noreply, new_state}
+  end
+
+  # Ignore all the other inputs
+  def handle_input(_input, _context, state), do: {:noreply, state}
+```
+
+☝️ We use the pattern matching to match the handler based on the received input.
+
+☝️ The [`put_in`](https://hexdocs.pm/elixir/Kernel.html#put_in/2) helper is used to update the `direction` in the scene's state.
+
+And here we are, now we can control the snake 🎉
```

Comment:
Maybe we can use a gif here for how the result should look like 🤔

Reply:
Absolutely! I added it, then maybe we will need to update it depending on your PR, let's see.

Reply:
Thanks 💚 

It's ace and I don't really expect the other PR to change the look too much. We eventually have to do a bit of resizing to the frame... But it shouldn't make a big difference for the gif.